### PR TITLE
Update Workers changelog to call out Iterator helpers

### PR DESCRIPTION
**Pull request by irvinebroque**
@garrettgu10

### CODE REVIEW

File: data/changelogs/workers.yaml [hunk 1]
```diff
@@ -2,6 +2,10 @@
 link: "/workers/platform/changelog/"
 productName: Workers
 entries:
+- publish_date: '2024-04-09'
+  description: |-
+    - v8 updated to version <x>
```
Comment (irvinebroque) [inline data/changelogs/workers.yaml]: @garrettgu10

Comment (garrettgu10) [reply data/changelogs/workers.yaml]: I may be missing some context here. The v8 docs say iterator helpers were added to 12.2, but we're currently in the process of rolling out 12.4

Comment (irvinebroque) [reply data/changelogs/workers.yaml]: that's what I needed to know — thanks :)

**Review by GregBrimble — APPROVED** [review]

File: data/changelogs/workers.yaml [hunk 2]
```diff
@@ -26,6 +26,10 @@ entries:
     - Built-in APIs that return Promises will now produce stack traces when the Promise rejects. Previously, the rejection error lacked a stack trace.
     - A new compat flag `fetcher_no_get_put_delete` removes the `get()`, `put()`, and `delete()` methods on service bindings and Durable Object stubs. This will become the default as of compatibility date 2024-03-26. These methods were designed as simple convenience wrappers around `fetch()`, but were never documented.
     - Updated v8 to version 12.3.
+- publish_date: '2024-02-24'
+  description: |-
+    - v8 updated to version 12.2
+    - You can now use [Iterator helpers](https://v8.dev/features/iterator-helpers) in Workers.
```
Comment (jasnell) [inline data/changelogs/workers.yaml:32]: Worth also mentioning that the new Set methods can be used now also. It's separate from iterator-helpers but also worth calling out. https://github.com/tc39/proposal-set-methods

**Review by jasnell — APPROVED** [review]

File: data/changelogs/workers.yaml [hunk 3]
```diff
@@ -26,6 +26,10 @@ entries:
     - Built-in APIs that return Promises will now produce stack traces when the Promise rejects. Previously, the rejection error lacked a stack trace.
     - A new compat flag `fetcher_no_get_put_delete` removes the `get()`, `put()`, and `delete()` methods on service bindings and Durable Object stubs. This will become the default as of compatibility date 2024-03-26. These methods were designed as simple convenience wrappers around `fetch()`, but were never documented.
     - Updated v8 to version 12.3.
+- publish_date: '2024-02-24'
+  description: |-
+    - v8 updated to version 12.2
```
Comment (deadlypants1973) [inline data/changelogs/workers.yaml]:
```suggestion
    - v8 updated to version 12.2.
```

**Review by deadlypants1973 — APPROVED** [review]

File: data/changelogs/workers.yaml (hunk 2 above)
Comment (irvinebroque) [reply data/changelogs/workers.yaml:32]:
```suggestion
    - You can now use [Iterator helpers](https://v8.dev/features/iterator-helpers) in Workers.
    - You can now use [new methods on `Set`](https://github.com/tc39/proposal-set-methods), such as `Set.intersection` and `Set.union`, in Workers.
```

File: data/changelogs/workers.yaml [hunk 4]
```diff
@@ -26,6 +26,11 @@ entries:
     - Built-in APIs that return Promises will now produce stack traces when the Promise rejects. Previously, the rejection error lacked a stack trace.
     - A new compat flag `fetcher_no_get_put_delete` removes the `get()`, `put()`, and `delete()` methods on service bindings and Durable Object stubs. This will become the default as of compatibility date 2024-03-26. These methods were designed as simple convenience wrappers around `fetch()`, but were never documented.
     - Updated v8 to version 12.3.
+- publish_date: '2024-02-24'
+  description: |-
+    - v8 updated to version 12.2.
+    - You can now use [Iterator helpers](https://v8.dev/features/iterator-helpers) in Workers.
+    - You can now use [new methods on `Set`](https://github.com/tc39/proposal-set-methods), such as `Set.intersection` and `Set.union`, in Workers.
```
Comment (irvinebroque) [inline data/changelogs/workers.yaml:33]: @jasnell curious why we got these in 12.2 but Node.js talks about these as part of 12.4?

https://nodejs.org/en/blog/announcements/v22-release-announce#v8-update-to-124

**Pull request merged**